electrum/wallet.py

```diff
@@ -1960,22 +1988,39 @@ def fee_estimator(size: Union[int, float, Decimal]) -> int:
             #       forever. see #5433
             # note: Actually, it might be the case that not all UTXOs from the wallet are
             #       being spent if the user manually selected UTXOs.
-            sendable = sum(map(lambda c: c.value_sats(), coins))
-            for (_,i) in i_max:
-                outputs[i].value = 0
+            def distribute_amount(amount):
+                if amount < 0:
+                    raise NotEnoughFunds()
+                distr_amount = 0
+                for (weight, i) in i_max:
+                    # fixme: this does not check that value >= dust_threshold
+                    val = int((amount/i_max_sum) * weight)
+                    outputs[i].value = val
+                    distr_amount += val
+                (x,i) = i_max[-1]
+                outputs[i].value += (amount - distr_amount)
+
+            distribute_amount(0)
             tx = PartialTransaction.from_io(list(coins), list(outputs))
```

Comment:
pre-existing bug on master from https://github.com/spesmilo/electrum/commit/8bec974a39a7692475db85b94c227b21a57bab31:
what about `inputs`? should those not be included in the spend max branch? unit test needed :)

Reply:
Good point. There is currently no exec path where inputs are passed to `make_unsigned_transaction` and we spend max. Maybe we can just add an `assert inputs == []` in the spend max branch?

Reply:
That works as a stopgap, yes. Though it is logically valid to try to spend max with some fixed inputs, you could just raise an exception saying it is not yet implemented.

Reply:
ok, now using `inputs + coins`. the refactoring of `should_keep_reserve_utxo` makes it possible